packages/optimizely-sdk/lib/optimizely/index.ts

```diff
@@ -0,0 +1,1383 @@
+          : DECISION_NOTIFICATION_TYPES.AB_TEST;
+
+        this.notificationCenter.sendNotifications(NOTIFICATION_TYPES.DECISION, {
+          type: decisionNotificationType,
+          userId: userId,
+          attributes: attributes || {},
+          decisionInfo: {
+            experimentKey: experimentKey,
+            variationKey: variationKey,
+          },
+        });
+
+        return variationKey;
+      } catch (ex) {
+        this.logger.log(LOG_LEVEL.ERROR, ex.message);
+        this.errorHandler.handleError(ex);
+        return null;
+      }
+    } catch (e) {
+      this.logger.log(LOG_LEVEL.ERROR, e.message);
+      this.errorHandler.handleError(e);
+      return null;
+    }
+  }
+
+  /**
+   * Force a user into a variation for a given experiment.
+   * @param  {string}      experimentKey
+   * @param  {string}      userId
+   * @param  {string|null} variationKey   user will be forced into. If null,
+   *                                      then clear the existing experiment-to-variation mapping.
+   * @return {boolean}                    A boolean value that indicates if the set completed successfully.
+   */
+  setForcedVariation(experimentKey: string, userId: string, variationKey: string | null): boolean {
+    if (!this.validateInputs({ experiment_key: experimentKey, user_id: userId })) {
+      return false;
+    }
+
+    const configObj = this.projectConfigManager.getConfig();
+    if (!configObj) {
+      return false;
+    }
+
+    try {
+      return this.decisionService.setForcedVariation(configObj, experimentKey, userId, variationKey);
+    } catch (ex) {
+      this.logger.log(LOG_LEVEL.ERROR, ex.message);
+      this.errorHandler.handleError(ex);
+      return false;
+    }
+  }
+
+  /**
+   * Gets the forced variation for a given user and experiment.
+   * @param  {string}      experimentKey
+   * @param  {string}      userId
+   * @return {string|null} The forced variation key.
+   */
+  getForcedVariation(experimentKey: string, userId: string): string | null {
+    if (!this.validateInputs({ experiment_key: experimentKey, user_id: userId })) {
+      return null;
+    }
+
+    const configObj = this.projectConfigManager.getConfig();
+    if (!configObj) {
+      return null;
+    }
+
+    try {
+      return this.decisionService.getForcedVariation(configObj, experimentKey, userId);
+    } catch (ex) {
+      this.logger.log(LOG_LEVEL.ERROR, ex.message);
+      this.errorHandler.handleError(ex);
+      return null;
+    }
+  }
+
+  /**
+   * Validate string inputs, user attributes and event tags.
+   * @param  {unknown}  stringInputs   Map of string keys and associated values
+   * @param  {unknown}  userAttributes Optional parameter for user's attributes
+   * @param  {unknown}  eventTags      Optional parameter for event tags
+   * @return {boolean}                 True if inputs are valid
+   *
+   */
+  private validateInputs(
+    stringInputs?: unknown,
```

Comment:
Looking at the callers, it appears we always pass an object. This will simplify the implementation:

```suggestion
    // Later we could make these camelCase, snake_case is weird
    stringInputs: Record<'feature_key' | 'user_id' | 'variable_key' | 'experiment_key', unknown>
```

Reply:
Made some modification to avoid missing properties compiler error and added `event_key`:
`Partial<Record<'feature_key' | 'user_id' | 'variable_key' | 'experiment_key' | 'event_key', unknown>>`